src/compile/repeat.ts

```diff
@@ -127,44 +132,39 @@ export class RepeatModel extends Model {
   }
 
   public parseScale(this: RepeatModel) {
-    const model = this;
-
     const scaleComponent: Dict<VgScale> = this.component.scales = {};
 
-    this.children.forEach(function(child) {
+    for (const child of this.children) {
       child.parseScale();
 
-      // FIXME(#1602): correctly implement independent scale
-      // Also need to check whether the scales are actually compatible, e.g. use the same sort or throw error
-      if (true) { // if shared/union scale
-        keys(child.component.scales).forEach(function(channel) {
-          if (contains(NONSPATIAL_SCALE_CHANNELS, channel)) {
-            const childScale = child.component.scales[channel];
-            const modelScale = scaleComponent[channel];
-
-            if (!childScale || isSignalRefDomain(childScale.domain) || (modelScale && isSignalRefDomain(modelScale.domain))) {
-              // TODO: merge signal ref domains
-              return;
-            }
-
-            if (modelScale) {
-              modelScale.domain = unionDomains(modelScale.domain, childScale.domain);
-            } else {
-              scaleComponent[channel] = childScale;
-            }
-
-            // rename child scale to parent scales
-            const scaleNameWithoutPrefix = childScale.name.substr(child.getName('').length);
-            const newName = model.scaleName(scaleNameWithoutPrefix, true);
-            child.renameScale(childScale.name, newName);
-            childScale.name = newName;
-
-            // remove merged scales from children
-            delete child.component.scales[channel];
+      // Check whether the scales are actually compatible, e.g. use the same sort or throw error
+      keys(child.component.scales).forEach((channel: ScaleChannel) => {
+        if (this.resolve[channel].scale === 'shared') {
+          const childScale = child.component.scales[channel];
+          const modelScale = scaleComponent[channel];
+
+          if (!childScale || isSignalRefDomain(childScale.domain) || (modelScale && isSignalRefDomain(modelScale.domain))) {
```

Comment:
FYI, I don't stand the logic for this line.  I know you are not really changing anything here, but this TODO things seems worth noting as an issue that we should deal with before 2.0?